activesupport/CHANGELOG.md

```diff
@@ -1,3 +1,41 @@
+*   Redis cache store.
+
+      # Defaults to `redis://localhost:6379/0`
+      config.cache_store = :redis_cache_store
+
+      # Supports all common cache store options (:namespace, :compress,
+      # :compress_threshold, :expires_in, :race_condition_tool) and all
+      # Redis options.
+      cache_password = Rails.application.secrets.redis_cache_password
+      config.cache_store = :redis_cache_store, driver: :hiredis,
+        namespace: 'myapp-cache', compress: true, timeout: 1,
+        url: "redis://:#{cache_password}@myapp-cache-1:6379/0"
+
+      # Supports Redis::Distributed with multiple hosts
+      config.cache_store = :redis_cache_store, driver: :hiredis
+        namespace: 'myapp-cache', compress: true,
+        url: %w[
+          redis://myapp-cache-1:6379/0
+          redis://myapp-cache-1:6380/0
+          redis://myapp-cache-2:6379/0
+          redis://myapp-cache-2:6380/0
+          redis://myapp-cache-3:6379/0
+          redis://myapp-cache-3:6380/0
+        ]
+
+      # Or pass a builder block
+      config.cache_store = :redis_cache_store,
+        namespace: 'myapp-cache', compress: true,
+        redis: -> { Redis.new … }
+
+    Deployment note: Take care to use a *dedicated Redis cache* rather
+    than pointing this at your existing Redis server. It won't cope well
+    with mixed usage patterns and it won't expire cache entries by default.
```

Comment:
Is this advice consistent with it defaulting to `redis://localhost:6379/0`? 🤔

Reply:
I initially set this up without a default for that reason, but it felt needlessly fussy for dev/test. I think clearly, repeatedly directing users to deployment guidance is the important ingredient.

Reply:
Maybe we should default to some non-nil expiry, and then people heeding the "how to use this at its best" guidance can turn that off at the same time they point to a dedicated instance?

Reply:
I like that direction since it require an informed choice of Redis usage mode.

Thing is, by default, Redis doesn't expire keys when it hits max mem—it returns errors. For expiry alone to work, we'd need to use `volatile-lru/ttl` max-mem policy instead of the default `noeviction`.

That suggests that we configure a volatile/allkeys policy on the cache store as well. Then we can inspect `maxmemory-policy` to satisfy ourselves that the server is on board with the user's intent.